mne/io/base.py

```diff
@@ -187,12 +203,33 @@ def to_data_frame(self, picks=None, index=None, scaling_time=1e3,
         for i, (k, v) in enumerate(mindex):
             df.insert(i, k, v)
         if index is not None:
-            if 'time' in index:
-                logger.info('Converting time column to int64...')
-                df['time'] = df['time'].astype(np.int64)
+            if 'time' in index and not long_format:
+                _set_pandas_dtype(df, ['time'], np.int64)
```

Comment:
any idea why we do this? it seems weird to me to force the time to be only at ms resolution.

@jona-sassenhagen any idea maybe?

Reply:
Nope ... But if we don't enforce, we need to store the unit somewhere right?

However, if time = 'ms', maybe that should be included in the column name anyways.

Reply:
>any idea why we do this? it seems weird to me to force the time to be only at ms resolution. 

It will break pandas indexing otherwise as you get non-unique values due to floating point noise. Old code Alex.

Reply:
ok then we can add parameter later if needed on time resolution.

Reply:
It's actually not common to use this for more than plotting / description. We currently have 3 digits precision which is reasonable I'd say. This PR does not touch this.